src/Iris/Cli/Interactive.hs

```diff
@@ -46,3 +49,8 @@ interactiveModeP = nonInteractiveP <|> pure Interactive
         [ Opt.long "no-input"
         , Opt.help "Enter the terminal in non-interactive mode"
         ]
+
+handleInteractiveMode :: InteractiveMode -> IO InteractiveMode
+handleInteractiveMode optionMode = do
+    supportsANSI <-  hSupportsANSI stdin
+    pure $ if supportsANSI then optionMode else NonInteractive
```

Comment:
I'd like to have tests for this function to make sure that it works on all 3 OSes.

Is there a way to check if we're running inside the GitHub actions environment? I imagine they should have some env variable we can check in tests to test the output of this function.

Otherwise, we can introduce our own env variable to test the output of this function on CI.

Reply:
I will add them. I didn't consider them earlier as I guess in a CI environment the terminal will always be noninteractive.
Still worth testing doesn't crash I guess

Reply:
Exactly. That's what we want to test. Something like:

```haskell
isCI <- checkCI
if isCI
then it "CI interactivity check" $ do
    handleInteractiveMode NonInteractive `shouldReturn` NonInteractive
    handleInteractiveMode Interactive `shouldReturn` NonInteractive
else it "Local interactivity check" $ do
    handleInteractiveMode NonInteractive `shouldReturn` NonInteractive
    handleInteractiveMode Interactive `shouldReturn` Interactive

Reply:
I implemented isCI by checking environment variable CI [set by github actions](https://docs.github.com/en/actions/learn-github-actions/environment-variables)